### PR TITLE
avm2: Make sure SoundChannel.soundTransform always returns a value

### PR DESCRIPTION
**Pull request by Dinnerbone**
This progresses Kingdom Rush Frontiers to the title screen.

### CODE REVIEW

File: core/src/avm2/globals/flash/media/soundchannel.rs [hunk 1]
```diff
@@ -88,15 +88,14 @@ pub fn sound_transform<'gc>(
...
-        .and_then(|this| this.as_sound_channel())
-        .and_then(|channel| channel.instance())
-    {
-        let dobj_st = activation.context.local_sound_transform(instance).cloned();
+    if let Some(channel) = this.and_then(|this| this.as_sound_channel()) {
+        let dobj_st = channel
+            .instance()
+            .and_then(|instance| activation.context.local_sound_transform(instance))
+            .cloned()
+            .unwrap_or_default();
```
Comment (adrian17) [inline core/src/avm2/globals/flash/media/soundchannel.rs:96]: unfortunately this is a bigger issue, looks like the transform is somehow bound to the channel, not to the instance?
```
	        var a = channel.soundTransform;
	        a.volume = 0.5;
	        channel.soundTransform = a;
	        var b = channel.soundTransform;
	        trace(a == b); // false, so the getter always returns a copy
	        trace(b.volume); // 0.5
```

**Comment by n0samu** [comment]
This fixes #8417! 🎉

**Comment by Dinnerbone** [comment]
It looks like this commit accidentally got lumped in with #9426 which got merged. I'll replace this PR with marking it as stubbed as we don't know what to do with the correct behaviour yet.

**Comment by Dinnerbone** [comment]
Dropping in favour of #9457

**Pull request closed without merging**